mtgjson4/provider/tcgplayer.py

```diff
@@ -146,22 +147,41 @@ def get_group_id_cards(group_id: int) -> List[Dict[str, Any]]:
     return cards
 
 
+def get_card_query_name(card: Dict[str, Any], set_code: str) -> str:
+    """
+    Returns the TCGPlayer-compatible query name. Removes diacritical marks.
+    Appends collector number to basic lands, and to GRN and RNA Guildgates.
+    :param card: MTGJSON card to find query name for
+    :param set_code: The code of the set the MTGJSON card belongs to
+    :return: TCGPlayer-compatible query name
+    """
+    unidecoded_name = unidecode.unidecode(card["name"])
+    if unidecoded_name in ["Plains", "Island", "Swamp", "Mountain", "Forest"]:
+        return "{} ({})".format(unidecoded_name, card["number"])
+    if set_code in ["GRN", "RNA"] and "Guildgate" in unidecoded_name:
```

Comment:
Is this really only a problem with these two sets? This function feels very hacky imo. Would it be better to decode all entries and try to fit them into TCGPlayer?

Reply:
It is a little hacky. Those are the only two sets where the Guildgates have variations. So you're maybe proposing something like trying both the original name and the name with collector number appended? That would encompass both the basic land and GRN/RNA Guildgate cases. It would probably help future proof as well. Then what about original vs. unidecoded name?

Reply:
Checking the data, there are currently no cases where a name with accents has its collector number appended in TCGPlayer. To help future proof, though, the number of checks could be expanded to 4: original name, appended name, unidecoded name, appended unidecoded name. This would eliminate the above function entirely. What do you think?